supportedBackends/jes/src/main/scala/cromwell/backend/impl/jes/JesAsyncBackendJobExecutionActor.scala

```diff
@@ -353,62 +384,84 @@ class JesAsyncBackendJobExecutionActor(override val standardParams: StandardAsyn
 
-    // In the event that the error isn't caught by the special cases, this will be used
-    def nonRetryableException = {
-      val exception = failed.toFailure(jobPaths.jobKey.tag, Option(jobPaths.stderr))
-      FailedNonRetryableExecutionHandle(exception, returnCode)
+    val msg = s"Retrying. $errorMessage"
+
+    previousRetryReasons match {
+      case Valid(PreviousRetryReasons(p, ur)) =>
+        val thisUnexpectedRetry = ur + 1
+        if (thisUnexpectedRetry <= maxUnexpectedRetries) {
+          // Increment unexpected retry count and preemption count stays the same
+          writeFuturePreemptedAndUnexpectedRetryCounts(p, thisUnexpectedRetry).map {
+            case _ => FailedRetryableExecutionHandle(StandardException(errorCode, msg, jobTag), jobReturnCode)
+          }
+        }
+        else {
+          Future.successful(FailedNonRetryableExecutionHandle(StandardException(errorCode, errorMessage, jobTag), jobReturnCode))
+        }
+      case Invalid(_) =>
+        Future.successful(FailedNonRetryableExecutionHandle(StandardException(errorCode, errorMessage, jobTag), jobReturnCode))
     }
+  }
+
+  private def handlePreemption(errorCode: Int, errorMessage: String, jobReturnCode: Option[Int]): Future[ExecutionHandle] = {
+    import lenthall.numeric.IntegerUtil._
 
-    val errorCode = failed.errorCode
-    val taskName = s"${workflowDescriptor.id}:${call.unqualifiedName}"
-    val attempt = jobDescriptor.key.attempt
-
-    failed.errorMessage match {
-      case Some(errorMessage) =>
-        if (errorMessage.contains("Operation canceled at"))  {
-          AbortedExecutionHandle
-        } else if (preempted(errorCode, errorMessage)) {
-          val preemptedMsg = s"Task $taskName was preempted for the ${attempt.toOrdinal} time."
-          if (attempt < maxPreemption) {
-            val e = PreemptedException(
-              s"""$preemptedMsg The call will be restarted with another preemptible VM (max preemptible attempts number is $maxPreemption).
-                 |Error code $errorCode. Message: $errorMessage""".stripMargin
-            )
-            FailedRetryableExecutionHandle(e, returnCode)
-          } else {
-            val e = PreemptedException(
-              s"""$preemptedMsg The maximum number of preemptible attempts ($maxPreemption) has been reached. The call will be restarted with a non-preemptible VM.
-                 |Error code $errorCode. Message: $errorMessage)""".stripMargin)
-            FailedRetryableExecutionHandle(e, returnCode)
+    previousRetryReasons match {
+      case Valid(PreviousRetryReasons(p, ur)) =>
+        val thisPreemption = p + 1
+        val taskName = s"${workflowDescriptor.id}:${call.unqualifiedName}"
+        val baseMsg = s"Task $taskName was preempted for the ${thisPreemption.toOrdinal} time."
+
+        writeFuturePreemptedAndUnexpectedRetryCounts(thisPreemption, ur).map { case _ =>
+          if (thisPreemption < maxPreemption) {
```

Comment:
It does work out, because when preemption is on the max value it retries non-preemptible, whereas when an unexpected retry is on the max value, we still want it to retry once more:

| This Preemption| Max Preemptions| Result  |
| ------------- |:-------------:| -----:|
|2|3| Retry preemptible |
|3|3| Retry non-preemptible |

| This Unexpected Retry| Max Unexpected Retries| Result  |
| ------------- |:-------------:| -----:|
|1|2| Retry |
|2|2| Retry |
|3|2| Fail |